common-cdi/src/main/java/fi/metatavu/edelphi/panels/PanelController.java

```diff
@@ -109,11 +131,60 @@ public void deletePanel(Panel panel) {
     panelUserGroupDAO.listAllByPanel(panel).forEach(panelUserGroupDAO::delete);
     panelUserExpertiseClassDAO.listByPanel(panel).forEach(panelUserExpertiseClassDAO::delete);
     panelUserIntressClassDAO.listByPanel(panel).forEach(panelUserIntressClassDAO::delete);
+
     panelStampDAO.listAllByPanel(panel).forEach(panelStampDAO::delete);
 
     panelDAO.delete(panel);
   }
 
+  /**
+   * Scheduler deletes these before deleting queries
+   *
+   * @param panel panel
+   */
+  public void schedulerDeleteQueryDependencies(Panel panel) {
```

Comment:
These should be separate methods (deletePanelAuths, deletePanelUserRoles, deletePanelBulletins and deletePanelInvitations)